src/services/apps-metadata.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { PredefinedProperties } from '../utils/types';
+
+// https://github.com/gridsuite/deployment/blob/main/docker-compose/docker-compose.base.yml
+// https://github.com/gridsuite/deployment/blob/main/k8s/resources/common/config/apps-metadata.json
+export type Metadata = MetadataCommon | MetadataStudy;
+export type Url = string | URL;
+
+export function fetchEnv() {
+    return fetch('env.json').then((res) => res.json());
+}
+
+export type MetadataCommon = {
+    name: string;
+    url: Url;
+    appColor: string;
+    hiddenInAppsMenu: boolean;
+};
+
+export type MetadataStudy = MetadataCommon & {
+    readonly name: 'Study';
+    resources?: {
+        types: string[];
+        path: string;
+    }[];
+    predefinedEquipmentProperties?: {
+        [networkElementType: string]: PredefinedProperties;
+    };
+    defaultParametersValues?: {
+        fluxConvention?: string;
+        enableDeveloperMode?: string; //maybe 'true'|'false' type?
+        mapManualRefresh?: string; //maybe 'true'|'false' type?
+    };
+    defaultCountry?: string;
+};
+
+export function fetchAppsMetadata(): Promise<Metadata[]> {
+    console.info(`Fetching apps and urls...`);
+    return fetchEnv()
+        .then((env) => fetch(env.appsMetadataServerUrl + '/apps-metadata.json'))
+        .then((response) => response.json());
+}
```

Comment:
```suggestion
export async function fetchAppsMetadata(): Promise<Metadata[]> {
    console.info(`Fetching apps and urls...`);
    return (await fetch((await fetchEnv()).appsMetadataServerUrl + '/apps-metadata.json')).json();
}
```

Reply:
We should keep more readable code.
```
const env = await fetchEnv();
const res = await fetch(env.appsMetadataServerUrl + '/apps-metadata.json')
return res.json()
```

Reply:
Yes one of the advantage of async / await is we can very easily split the code